library/Closed/Internal.hs

```diff
@@ -242,3 +242,10 @@ multiply (Closed x) (Closed y) = Closed $ x * y
 isValidClosed :: (KnownNat n, KnownNat m) => Closed n m -> Bool
 isValidClosed cx@(Closed x) =
   natVal (lowerBound cx) <= x && x <= natVal (upperBound cx)
+
+-- | Clamp an @'Integral'@ in the range constrained by a @'Closed'@ interval
+clamp :: forall n m a. (KnownNat n, KnownNat m, n <= m, Integral a) => a -> Closed n m
+clamp x
+  | fromIntegral x < getClosed (minBound @(Closed n m)) = minBound
+  | fromIntegral x > getClosed (maxBound @(Closed n m)) = maxBound
+  | otherwise = Closed (fromIntegral x)
```

Comment:
Nice! Can you move it up with the other constructors (`closed`, `unsafeClosed`) and add an example/test to `README.md`?

Reply:
Done! I was wondering where the tests were. Does hspec-discover handle running these?

Reply:
Nah, we have a symbolic link to `README.md` called `README.lhs` and we treat that as the `Main` module for the test stanza. [See here](https://github.com/freckle/closed/blob/master/package.yaml#L28)